pkg/services/sqlstore/migrations/ualert/ualert.go

```diff
@@ -435,3 +415,81 @@ type rmMigrationWithoutLogging = rmMigration
 func (m *rmMigrationWithoutLogging) SkipMigrationLog() bool {
 	return true
 }
+
+type upgradeNgAlerting struct {
+	migrator.MigrationBase
+}
+
+var _ migrator.CodeMigration = &upgradeNgAlerting{}
+
+func (u *upgradeNgAlerting) Exec(sess *xorm.Session, migrator *migrator.Migrator) error {
+	//if there are records with org_id == 0 then the feature flag was enabled before 8.2 that introduced org separation, and actually added the column.
+	//if feature is enabled in 8.2 the migration "AddDashAlertMigration", which is effectively different from what was run in 8.1.x and earlier versions,
+	//will handle organizations correctly, and, therefore, nothing needs to be fixed
+	count, err := sess.Table(&AlertConfiguration{}).Where("org_id = 0").Count()
+	if err != nil {
+		return fmt.Errorf("failed to query table alert_configuration: %w", err)
+	}
+	if count == 0 {
+		return nil //NOTHING TO DO
+	}
+
+	orgs := make([]int64, 0)
+	//get all org IDs sorted in ascending order
+	if err = sess.Table("org").OrderBy("id").Cols("id").Find(&orgs); err != nil {
+		return fmt.Errorf("failed to query table org: %w", err)
+	}
+	if len(orgs) == 0 { //should not really happen
+		return fmt.Errorf("no organizations found")
+	}
+	if len(orgs) == 1 {
+		firstOrg := orgs[0]
+		migrator.Logger.Info("Assigning all records from alert_configuration to the first organization", "org", firstOrg)
+		_, err = sess.Cols("org_id").Update(&AlertConfiguration{OrgID: firstOrg})
+		if err != nil {
+			return fmt.Errorf("failed to update org_id for all rows in the table alert_configuration: %w", err)
+		}
+		return nil
+	}
+
+	migrator.Logger.Info("Detected many organizations. The current alertmanager configuration will be replaced by the default one")
+
+	configs := make([]*AlertConfiguration, 0, len(orgs))
+	for _, org := range orgs {
+		configs = append(configs, &AlertConfiguration{
+			AlertmanagerConfiguration: `{
+	"alertmanager_config": {
+		"route": {
+			"receiver": "grafana-default-email"
+		},
+		"receivers": [{
+			"name": "grafana-default-email",
+			"grafana_managed_receiver_configs": [{
+				"uid": "",
+				"name": "email receiver",
+				"type": "email",
+				"isDefault": true,
+				"settings": {
+					"addresses": "<example@email.com>"
+				}
+			}]
+		}]
+	}
+}`,
+			// Since we are migration for a snapshot of the code, it is always going to migrate to
+			// the v1 config.
+			ConfigurationVersion: "v1",
+			OrgID:                org,
+		})
+	}
+
+	_, err = sess.InsertMulti(configs)
+	if err != nil {
+		return fmt.Errorf("failed to add default alertmanager configurations to every organization: %w", err)
+	}
+	return nil
```

Comment:
Following the previous comment, about the silence file:
`<data_path>/alerting/silences`
It would be good to clean it, even though it's not harmful since is no longer used.